-- Macro to call a function returning its first result when successful,
-- otherwise raise an error prefixed with the faild function name.
## function fn_assert(v)
  local expr_ret, expr_err: string = #[v]#
  if #expr_err ~= 0 then
    ## local funcname = context.scope:get_up_scope_of_kind('is_function').funcsym.name
    error(#[funcname .. ': ']# .. expr_err)
  end
  in expr_ret
## end
